firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/internal/InstallActivity.java

```diff
@@ -53,16 +61,15 @@ protected void onResume() {
 
     if (!isUnknownSourcesEnabled()) {
       // See comment about install progress above. Same applies to unknown sources UI.
-      if (unknownSourceEnablementInProgress) {
+      if (alertDialog.isShowing()) {
         LogWrapper.getInstance()
             .e(
                 TAG
                     + "Unknown sources enablement was already in progress. It was either canceled or failed");
+        alertDialog.dismiss();
```

Comment:
Ah, does this avoid a "window leaked error" or something?

Reply:
Yes it seems that we were not dismissing the dialog as this gets destroyed and so before we finish I decided to dismiss the dialog.

Reply:
What happens if the Dialog is showing and we call finish()? (for example, if the if statement above is triggered). Do we need to dismiss the dialog in the onDestroy method of this action, too?

Reply:
This is being called before finish so that should be ok. A scenario I am testing now is if the activity gets destroyed.

Reply:
What if we trigger the finish() on line 50 while the dialog is showing?